File: app/src/main/java/com/bubelov/coins/api/coins/NewUserParams.java
package com.bubelov.coins.api.coins;

import java.util.HashMap;
import java.util.Map;

/**
 * @author Igor Bubelov
 */

public class NewUserParams {
    private final Map<String, Object> user;

    public NewUserParams(String email, String password, String firstName, String lastName) {
        user = new HashMap<>();
        user.put("email", email);
        user.put("password", password);
        user.put("first_name", firstName);
        user.put("last_name", lastName);
    }
}

File: app/src/main/java/com/bubelov/coins/api/coins/NewUserParams.kt
package com.bubelov.coins.api.coins

import java.util.HashMap

/**
 * @author Igor Bubelov
 */

class NewUserParams(email: String, password: String, firstName: String, lastName: String) {
    private val user: MutableMap<String, Any>

    init {
        user = HashMap<String, Any>().apply {
            put("email", email)
            put("password", password)
            put("first_name", firstName)
            put("last_name", lastName)
        }
    }
}